File: backend/models/Report.js
const mongoose = require('mongoose');
const mongoosePaginate = require('mongoose-paginate-v2');

const reportSchema = new mongoose.Schema({
  title: {
    type: String,
    required: [true, 'Title is required'],
    trim: true,
    maxlength: [150, 'Title cannot exceed 150 characters'],
  },
  description: {
    type: String,
    trim: true,
    maxlength: [1000, 'Description cannot exceed 1000 characters'],
  },
  category: {
    type: String,
    enum: [
      'pothole', 'road_crack', 'open_manhole', 'flooding', 'other',
      'FallenTrees', 'DamagedElectricalPoles', 'Potholes and RoadCracks', 'Garbage',
      'pothole_crack', 'garbage', 'fallen_tree', 'poles'
    ],
    required: [true, 'Category is required'],
  },
  imageUrl: {
    type: String,
    default: '',
  },
  location: {
    type: {
      type: String,
      default: 'Point',
      enum: ['Point'],
    },
    coordinates: {
      type: [Number], // [longitude, latitude]
      required: [true, 'Coordinates are required'],
    },
  },
  address: {
    type: String,
    default: '',
    trim: true,
  },
  state: {
    type: String,
    default: '',
    trim: true,
  },
  status: {
    type: String,
    enum: ['pending', 'in_progress', 'resolved'],
    default: 'pending',
  },
  severity: {
    type: Number,
    min: [1, 'Severity must be at least 1'],
    max: [5, 'Severity cannot exceed 5'],
    default: 3,
  },
  upvotes: {
    type: Number,
    default: 0,
  },
  upvotedBy: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
    },
  ],
  gravityScore: {
    type: Number,
    default: 0,
  },
  isDuplicate: {
    type: Boolean,
    default: false,
  },
  mergedWith: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Report',
    default: null,
  },
  submittedBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  assignedTo: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    default: null,
  },
  aiValidated: {
    type: Boolean,
    default: false,
  },
  aiConfidence: {
    type: Number,
    default: null,
  },
  aiBoundingBox: {
    type: [Number],
    default: null,
  },
  estimatedDays: {
    type: Number,
    default: null,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

// 2dsphere index for geospatial queries
reportSchema.index({ location: '2dsphere' });

// Paginate plugin
reportSchema.plugin(mongoosePaginate);

module.exports = mongoose.model('Report', reportSchema);

File: spring-backend/src/main/java/com/civica/model/Report.java
package com.civica.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.GeoSpatialIndexType;
import org.springframework.data.mongodb.core.index.GeoSpatialIndexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "reports")
public class Report {

    @Id
    private String id;

    private String title;
    private String description;
    private String category;
    private String imageUrl;

    @GeoSpatialIndexed(type = GeoSpatialIndexType.GEO_2DSPHERE)
    private GeoJsonPoint location;

    @Builder.Default
    private String address = "";

    @Builder.Default
    private String state = "";

    @Builder.Default
    private String status = "pending";

    @Builder.Default
    private Integer severity = 3;

    @Builder.Default
    private Integer upvotes = 0;

    @Builder.Default
    private List<String> upvotedBy = new ArrayList<>();

    @Builder.Default
    private Double gravityScore = 0.0;

    @Builder.Default
    private Boolean isDuplicate = false;

    private String mergedWith;

    private String submittedBy;

    private String assignedTo;

    @Builder.Default
    private Boolean aiValidated = false;

    private Double aiConfidence;

    private List<Double> aiBoundingBox;

    private Integer estimatedDays;

    @Builder.Default
    private Instant createdAt = Instant.now();
}
